discussment-core/src/main/java/org/danekja/discussment/core/dao/jpa/GenericDaoJPA.java

```diff
@@ -22,16 +22,19 @@ public GenericDaoJPA(Class<T> clazz, EntityManager em) {
     }
 
     public T save(T obj) {
-        if(!em.getTransaction().isActive()) {
+        boolean isSpringActive = em.isJoinedToTransaction();
```

Comment:
Doesnt necessarily mean Spring is active, simply checks for existing transaction. Rename to hasOuterTransaction